src/robot_dart/sensor/torque.cpp

```diff
@@ -0,0 +1,134 @@
+#include "torque.hpp"
+
+#include <robot_dart/robot_dart_simu.hpp>
+
+#include <dart/dynamics/BodyNode.hpp>
+#include <dart/dynamics/Joint.hpp>
+
+#include <dart/dynamics/BallJoint.hpp>
+#include <dart/dynamics/EulerJoint.hpp>
+#include <dart/dynamics/FreeJoint.hpp>
+#include <dart/dynamics/PlanarJoint.hpp>
+#include <dart/dynamics/PrismaticJoint.hpp>
+#include <dart/dynamics/RevoluteJoint.hpp>
+#include <dart/dynamics/ScrewJoint.hpp>
+#include <dart/dynamics/TranslationalJoint.hpp>
+#include <dart/dynamics/TranslationalJoint2D.hpp>
+#include <dart/dynamics/UniversalJoint.hpp>
+#include <dart/dynamics/WeldJoint.hpp>
+
+namespace robot_dart {
+    namespace sensor {
+        Torque::Torque(RobotDARTSimu* simu, dart::dynamics::Joint* joint, size_t frequency, const std::string& direction)
+        :   Sensor(simu, frequency),
+            _direction(direction),
+            _torques(joint->getNumDofs()),
+            _motion_subspace(Eigen::MatrixXd(6, joint->getNumDofs()))
+        {
+            attach_to_joint(joint, Eigen::Isometry3d::Identity());
+            update_motion_subspace(joint);
+        }
+
+        void Torque::init()
+        {
+            _wrench.setZero();
+            _torques.setZero();
+
+            attach_to_joint(_joint_attached, Eigen::Isometry3d::Identity());
+            _active = true;
+        }
+
+        void Torque::calculate(double t)
+        {
+            if (!_attached_to_joint)
+                return; // cannot compute anything if not attached to a joint
+
+            Eigen::Vector6d F2 = Eigen::Vector6d::Zero();
+            auto child_body = _joint_attached->getChildBodyNode();
+            // ROBOT_DART_ASSERT(child_body != nullptr, "Child BodyNode is nullptr", );
+            if (child_body)
+                F2 = -dart::math::dAdT(_joint_attached->getTransformFromChildBodyNode(), child_body->getBodyForce());
+
+            // We always compute things in SENSOR frame (aka joint frame)
+            _wrench = (_direction == "parent_to_child") ? F2 : -F2;
+
+            Eigen::Matrix3d R = _joint_attached->getTransformFromChildBodyNode().linear().transpose();
```

Comment:
Why do we need this? F2 is already in joint space. We need it back in `ChildBodyNode` coordinates? Why?

Reply:
the implementation was based on the FT sensor class.
yes line 53 and below should be useless for the torque sensor